Fix Unit construction with no arguments (#11295)
astropy/astropy: CHANGES.rst

- Make ufunc helper lookup thread-safe. [#11226]

> - Fix ``Unit()`` argument-less constructor, returning the null Unit. 
>   It was documented but not implemented. [#11295]

Reviewer: ```suggestion
- Calling ``Unit()`` with no argument now returns a dimensionless unit, 
  as was documented but not implemented. [#11295]
```

Reviewer: Also, this might require a note under API changes as well? @pllim for something like this, does it make sense to have in two places in the changelog?

Reviewer 2: Yes, there is a precedence that the same PR has multiple entries, though it does not make sense if the second entry is a copy-paste of the first one; usually they each explain different parts of the PR. Hope this makes sense!

Reviewer 3: I think this makes more sense under "API change" - most users will probably not have run into this as a bug (agree with the suggested change by @adrn for the same reason). Also, it will give it a bit more visibility! (but I don't think this needs a what's-new).
